db/compaction/compaction_iterator.cc

```diff
@@ -1280,23 +1280,27 @@ void CompactionIterator::DecideOutputLevel() {
   }
 #endif  // NDEBUG
 
+  // saved_seq_for_penul_check_ is populated in `NextFromInput` when the
+  // entry's sequence number is non zero and validity context for output this
+  // entry is kSwapPreferredSeqno for use in `DecideOutputLevel`. It should be
+  // cleared out here unconditionally.
```

Comment:
It wasn't immediately clear to me that the field is cleared because this is where the value is consumed.

Reply:
Yeah, I added this loud comment to emphasize this.

Reply:
I mean reading your comment didn't make it clear. I went searching for other uses of `saved_seq_for_penul_check_` to get more context on why "it should be cleared out here."

Reply:
I see, I added some clarification, specifically it's this: "Otherwise, it may end up getting consumed
  incorrectly by a different entry"